invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/landing_page/ShareOptions/AccessUsers/AddUserAccessModal.js

```diff
@@ -0,0 +1,128 @@
+import { SearchWithRoleSelection } from "@js/invenio_communities/members";
+import { RichEditor } from "react-invenio-forms";
+import { GrantAccessApi } from "../api/api";
+
+export class AddUserAccessModal extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { open: false, notifyUser: true, message: undefined };
+  }
+
+  onSuccess = () => {
+    const { fetchData } = this.props;
+    fetchData();
+    this.handleCloseModal();
+  };
+
+  updateMessage = (message) => {
+    this.setState({ message: message });
+  };
+
+  handleNotifyCheckboxClick = () => {
+    const { notifyUser } = this.state;
+    this.setState({ notifyUser: !notifyUser });
+  };
+
+  handleOpenModal = () => this.setState({ open: true });
+  handleCloseModal = () => {
+    this.setState({ open: false, notifyUser: true });
+  };
+
+  render() {
+    const { results, record, isComputer, accessDropdownOptions } = this.props;
+    const { open, notifyUser, message } = this.state;
+
+    const api = new GrantAccessApi(record);
+
+    const existingIds = [];
+    results.map((result) => {
+      existingIds.push(result?.subject?.id);
+    });
+
+    return (
+      <Modal
+        role="dialog"
+        onClose={this.handleCloseModal}
+        onOpen={this.handleOpenModal}
+        closeOnDimmerClick={false}
+        open={open}
+        aria-label={i18next.t("Add people")}
+        trigger={
+          <Button
+            className={!isComputer ? "mobile only tablet only mb-15" : ""}
+            content={i18next.t("Add people")}
+            positive
+            size="medium"
+            icon="plus"
+            labelPosition="left"
+            floated={!isComputer ? "right" : undefined}
+          />
+        }
+      >
+        <Modal.Header as="h2">{i18next.t("Add people")}</Modal.Header>
+        <SearchWithRoleSelection
+          key="access-users"
+          roleOptions={accessDropdownOptions}
+          modalClose={this.handleCloseModal}
+          action={api.createGrants}
+          onSuccessCallback={this.onSuccess}
+          searchBarTitle={<label>{i18next.t("User")}</label>}
+          searchBarTooltip={i18next.t(
+            'Users must set profile visibility to "Public" in order to be granted access.'
+          )}
+          doneButtonText={i18next.t("Add")}
+          doneButtonIcon="plus"
+          radioLabel={i18next.t("Access")}
+          selectedItemsHeader={i18next.t("Selected users")}
+          message={message}
+          messageComponent={
+            <>
+              <Checkbox
+                checked={notifyUser}
+                className="mb-20 mt-10"
+                label={i18next.t("Notify people")}
+                onClick={this.handleNotifyCheckboxClick}
+              />
+              {notifyUser && (
+                <>
+                  <p>
+                    <b>{i18next.t("Message")}</b>
+                  </p>
+                  <RichEditor
+                    onBlur={(event, editor) => {
```

Comment:
I think that (to be checked) it you click on the submit/add button just after writing, the `onBlur` is not triggered, but I am not 100% sure.
In some conditions, the `onBlur` was not enough.

Reply:
Checked, it does get triggered even if you click `Submit` right after writing the message